Collections/Bag/Bag.swift

```diff
@@ -66,15 +66,16 @@ struct Bag<ContainedElement: Hashable> {
         }
     }
 
-    /// Removes occurences times the specific member. If occurences < 0, all occurences of this member get removed
+    /// Removes occurences times the specific member.
     mutating func remove(_ member: ContainedElement, occurrences: Int = 1) {
+        precondition(occurrences > 0, "Can only remove a positive number of occurrences")
+        
         let currentCount = contents[member] ?? 0
-
-        if occurrences >= 0 && currentCount > occurrences {
-            contents[member] = currentCount - occurrences
-        } else {
-            contents.removeValue(forKey: member)
-        }
+        contents[member] = currentCount - occurrences
```

Comment:
Please add handling for elements that are currently not contained. Otherwise a call would result in value of -1 for the given key.

Reply:
Fixed and improved.